packages/headless/src/features/analytics/analytics-utils.ts

```diff
@@ -170,5 +174,15 @@ function partialResultPayload(result: Result): Partial<Result> {
   );
 }
 
+function getDocumentAuthor(r: Result): string {
+  if (isNullOrUndefined(r.raw['author'])) {
+    return 'unknown';
+  }
+
+  return Array.isArray(r.raw['author'])
+    ? r.raw['author'].join(';')
+    : (r.raw['author'] as string);
+}
```

Comment:
We could store `r.raw['author']` in a variable to reduce the repetition. Also, could we avoid the cast by
a. Better typing `author` on the `Result` interface, or if not possible,
b. Marking the variable as `unknown`, and using a template string to cast ints, booleans to strings?

```
const author = r.raw['author'];

if (isNullOrUndefined(...)) {
    return 'unknown'
}

return Array.isArray(...) ? author.join(';') : `${author}`; // casts non-string types
```

Reply:
A small addition to that could be to use destructuring in the parameter since `author` is an object key and its value is the only part of `r` used in this function:
`function getDocumentAuthor({ raw: { author } }: Result): string { ... }